drivers/stepper/adi_tmc/adi_tmc5041_stepper_controller.c

```diff
@@ -103,20 +105,22 @@ static void calculate_velocity_from_hz_to_fclk(const struct device *dev, const u
 		velocity_hz, *velocity_fclk);
 }
 
-static void set_async_signal(const struct device *dev, struct k_poll_signal *async)
+#ifdef CONFIG_STEPPER_ASYNC_API
+
+static int tmc5041_stepper_set_event_callback(const struct device *dev,
+					      stepper_event_callback_t callback)
 {
 	struct tmc5041_stepper_data *data = dev->data;
 
-	if (!async) {
-		return;
-	}
-
-	if (data->async_signal) {
-		k_poll_signal_reset(data->async_signal);
+	if (!callback) {
+		return -EINVAL;
```

Comment:
imo it should be possible to unset the callback

Reply:
Hmm what would be the usecase? You either want to process events or not, and i think this won't change during runtime, or?

The decision of processing events or not should be left to the user imo, or?